### ISSUE TITLE
Incorrect behavior of ExplicitBoolCompareRector, CountArrayToEmptyArrayComparisonRector

### ISSUE THREAD

**Issue by lptn** [rectorphp/rector#6645]
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/1ebff836-affa-60c0-bb5f-a1e7ad062396

```php
<?php declare(strict_types=1);

namespace App\Console\Commands\Infrastructure\ServerAuditInspections;

final class PhpCliInspection
{
    private array $variablesToCheck = [
        'key' => ['=' => ''],
    ];

    public function inspect(): void
    {
        $problems = [];

        foreach ($this->variablesToCheck as $variableName => $comparisons) {
            if ($comparisons !== null) {
                    $problems[] = $variableName;
                }
        }

        if (count($problems)) {
            throw new \Exception();
        }
    }

    private function check(string $variableName, string $operator, string $expectedValue): ?string
    {
        return array_rand(['a', null], 1);
    }
}
```

### Responsible rules

* `ExplicitBoolCompareRector`

* `CountArrayToEmptyArrayComparisonRector`

## Expected Behavior

`if ($problems !== []) {`

**Comment by samsonasik** [comment]
should already fixed at https://github.com/rectorphp/rector-src/pull/474

Please try latest 0.11.48

**Comment by samsonasik** [comment]
could you add more fixture under https://github.com/rectorphp/rector-src/tree/main/tests/Issues/Issue6561/Fixture

thank you.

**Comment by lptn** [comment]
>should already fixed at rectorphp/rector-src#474

thanks!

>Please try latest 0.11.48

unfortunately it crashes on my project (0.11.47 works fine):

```
In BreakingRemovalGuard.php line 25:

  Node "PhpParser\Node\Expr\StaticCall" on line 133 is child of "PhpParser\No
  de\Expr\Assign", so it cannot be removed as it would break PHP code. Change
   or remove the parent node instead.

process [-n|--dry-run] [-a|--autoload-file AUTOLOAD-FILE] [-o|--output-format [OUTPUT-FORMAT]] [--no-progress-bar] [--no-diffs] [--clear-cache] [--] [<source>...]

Script rector process handling the rector event returned with error code 1
Error: Process completed with exit code 1.
```

**Comment by samsonasik** [comment]
please provide failing test case for that, thank you.
